x-pack/plugins/endpoint/public/embeddables/resolver/store/camera/reducer.ts

```diff
@@ -59,118 +65,145 @@ export const cameraReducer: Reducer<CameraState, ResolverAction> = (
         state.latestFocusedWorldCoordinates,
-        projectionMatrix(state)
+        selectors.projectionMatrix(state)(action.payload.time)
+      );
+      const newWorldCoordinatesAtLastFocusedPosition = vector2.applyMatrix3(
+        rasterOfLastFocusedWorldCoordinates,
+        selectors.inverseProjectionMatrix(stateWithNewScaling)(action.payload.time)
+      );
+
+      /**
+       * The change in world position incurred by changing scale.
+       */
+      const delta = vector2.subtract(
+        newWorldCoordinatesAtLastFocusedPosition,
+        state.latestFocusedWorldCoordinates
       );
-      const matrix = inverseProjectionMatrix(stateWithNewScaling);
-      const worldCoordinateThereNow = applyMatrix3(rasterOfLastFocusedWorldCoordinates, matrix);
-      const delta = subtract(worldCoordinateThereNow, state.latestFocusedWorldCoordinates);
 
-      return {
+      /**
+       * Adjust for the change in position due to scale.
+       */
+      const translationNotCountingCurrentPanning: Vector2 = vector2.subtract(
+        stateWithNewScaling.translationNotCountingCurrentPanning,
+        delta
+      );
+
+      const nextState: CameraState = {
         ...stateWithNewScaling,
-        translationNotCountingCurrentPanning: [
-          stateWithNewScaling.translationNotCountingCurrentPanning[0] + delta[0],
-          stateWithNewScaling.translationNotCountingCurrentPanning[1] + delta[1],
-        ],
+        translationNotCountingCurrentPanning,
       };
+
+      return nextState;
     } else {
       return stateWithNewScaling;
     }
   } else if (action.type === 'userSetPositionOfCamera') {
     /**
      * Handle the case where the position of the camera is explicitly set, for example by a 'back to center' feature.
      */
-    return {
+    const nextState: CameraState = {
       ...state,
+      animation: undefined,
       translationNotCountingCurrentPanning: action.payload,
     };
+    return nextState;
   } else if (action.type === 'userStartedPanning') {
+    if (selectors.isAnimating(state)(action.payload.time)) {
+      return state;
+    }
     /**
      * When the user begins panning with a mousedown event we mark the starting position for later comparisons.
      */
-    return {
+    const nextState: CameraState = {
       ...state,
+      animation: undefined,
       panning: {
-        origin: action.payload,
-        currentOffset: action.payload,
+        origin: action.payload.screenCoordinates,
+        currentOffset: action.payload.screenCoordinates,
       },
     };
+    return nextState;
   } else if (action.type === 'userStoppedPanning') {
     /**
      * When the user stops panning (by letting up on the mouse) we calculate the new translation of the camera.
      */
-    if (userIsPanning(state)) {
-      return {
+    if (state.panning) {
+      const nextState: CameraState = {
         ...state,
-        translationNotCountingCurrentPanning: translation(state),
+        /**
```

Comment:
TODO why?

Reply:
can we just pass `time` to `userStoppedPanning` and use the normal function? would simplify types